core/src/avm2/object/loaderinfo_object.rs

```diff
@@ -147,17 +169,52 @@ impl<'gc> TObject<'gc> for LoaderInfoObject<'gc> {
                     e
                 );
             }
+
+            let complete_evt = EventObject::bare_default_event(context, "complete");
+
+            if let Err(e) = Avm2::dispatch_event(context, complete_evt, (*self).into()) {
+                log::error!(
+                    "Encountered AVM2 error when broadcasting `complete` event: {}",
```

Comment:
side note: maybe we should push this into `dispatch_event`? As it's currently duplicated many times already, I think

Reply:
Yeah, I was thinking about doing that in a follow-up PR.